tests/test-int.cpp

```diff
@@ -41,6 +41,27 @@ class IntTests : public ::testing::Test,
   Sort intsort;
 };
 
+// test for creating variadic mult-terms
+TEST_P(IntTests, Mult)
+{
+  Term two = s->make_term(2, intsort);
+  Term three = s->make_term(3, intsort);
+  Term four = s->make_term(4, intsort);
+  Term twentyfour = s->make_term(24, intsort);
+  Term mult;
+  try
+  {
+    mult = s->make_term(Mult, {two, three, four});
+  }
+  catch (const IncorrectUsageException &e)
```

Comment:
Perhaps we could try this without catching? This will tell us what solvers have problems with that, and then we can fix it for those solvers?

Ideally we would like to be uniform w.r.t. back-end solvers, as much as possible.

Reply:
I'm not sure what you mean. If an exception is thrown, then the FAIL in the catch-block is executed, and the solver's error message is shown:

`FAIL() << "creating mult-term failed: " << e.what();`

For example, if I execute the test in the master branch on my machine, then I get:

```
$SMT_SWITCH/tests/test-int.cpp:58: Failure
Failed
creating mult-term failed: Can't apply * to 3 terms.
[  FAILED  ] ParameterizedSolverIntTests/IntTests.Mult/4, where GetParam() = Z3(logging=0) (1 ms)
```

Since all CI checks pass with my fix, I guess that Z3 is the only solver that can't handle variadic multiplication.

I guess it would also work as desired without catching (I'm not very familiar with gtest and how it handles exceptions), but then we cannot provide a custom error message ("creating mult-term failed: "...).

Reply:
Sorry, you are right, I missed the fact that the message goes to `FAIL()`. So if `z3` is the only solver for which this does not work, could you perhaps try to make it work for that too? It should support it so it is unclear why this does not work:

https://github.com/stanford-centaur/smt-switch/blob/f4bd379333f06955eb147ac419b8d1f3fdd0d6fd/z3/src/z3_solver.cpp#L99

Reply:
With my fix (a757315), it does work with Z3 (which is the reason why the CI checks passed after my last commit 61b4855). The problem was that `z3_solver` only tries to build a variadic term if the corresponding operation is marked as variadic in `ops.h`, which was not the case for `Mult`.